driver-core/src/main/com/mongodb/client/model/expressions/Expression.java

```diff
@@ -21,12 +21,68 @@
+ * on a MongoDB server or compatible execution context.
+ *
+ * <p>The following is an example of usage within an aggregation pipeline. Here,
+ * the current document value is obtained and its "numberArray" field is
+ * filtered and summed, in a style similar to that of the Java Stream API:
+ *
+ * <pre>{@code
+ * AggregateIterable<Document> result = col.aggregate(Arrays.asList(
+ *     addFields(new Field<>("result", current()
+ *         .<MqlNumber>getArray("numberArray")
+ *         .filter(v -> v.gt(of(0)))
+ *         .sum(v -> v)))));
+ * }</pre>
+ *
+ * <p>Values are typically obtained via the current document and its fields, or
+ * specified via statically-imported methods on the {@link Expressions} class.
+ *
+ * <p>The null value is not part of, and cannot be used as if it were part of,
+ * any other type. It has no explicit type of its own. Instead of checking for
+ * null, users should check for their expected type, via methods such as
+ * {@link Expression#isNumberOr(NumberExpression)}. Where the null value must
+ * be checked explicitly, users may use {@link Branches#isNull} within
+ * {@link Expression#switchOn}.
+ *
+ * <p>There is no explicit "missing" or "undefined" value. Users may use
+ * {@link DocumentExpression#has}.
+ *
+ * <p>This type hierarchy differs from BSON in that it provides operations,
+ * while the BSON API does not, its numeric types are less granular, and it
+ * offers multiple abstractions of certain types (document, map, entry). It
+ * differs from Java types in that the operations available differ, and in that
+ * an implementation of this API may be used to be used to generate MQL in the
+ * form of BSON. (This API makes no guarantee regarding the BSON output produced
+ * by its implementation, which in any case may vary due to optimization or
+ * other factors.)
+ *
+ * <p>This API is a language binding in the sense that it exposes the operations
+ * available on the MongoDB server or compatible execution context, and the
+ * implied corresponding data types, through a Java API. The methods exposed
+ * through this API correspond to MQL operations, though this correspondence
+ * is not exact.
+ *
+ * <p>Some methods within the API constitute an assertion by the user that the
+ * data is of a certain type. For example, {@link DocumentExpression#getArray}}
+ * requires that the underlying field is both an array, and an array of some
+ * certain type. If the field is not an array in the underlying data, behaviour
+ * is undefined by this API (though behaviours may be defined by the execution
+ * context, users are strongly discouraged from relying on behaviour that is not
+ * part of this API).
+ *
+ * <p>As with the Java Stream API's terminal operations, corresponding Java
+ * values are not directly available, but must be obtained indirectly via the
+ * aggregation pipeline, or via find. Certain methods may cause an error, which
+ * will be produced through these "terminal operations".
```

Comment:
- This section should probably be moved up directly before _"The following is an example of usage within an aggregation pipeline ..."_
- _"via the aggregation pipeline, or via find"_ - we should link to the relevant Java APIs.

Reply:
How could I link those APIs from here?

Reply:
I moved it after the section discussing "initial" operations/expressions.

Reply:
[marked as optional]

> How could I link those APIs from here?

Then code dependencies don't allow me to use `@link`, I use `{@code MongoCollection.aggregate}`/`{@code MongoCollection.find}`.